File: examples/test-terry-davis.py
#!/usr/bin/env python3
"""
Test of terry-davis skill - Python version
This demonstrates the Terry Davis coding style in Python
"""

# God said this should work
# The best code is the code you never wrote... but this is necessary

import random

# Cosmic variable names
GodPointer = None
DivineArray = [1, 2, 3, 4, 5]
HolyCVariable = 42
TempleOSKernel = "TempleOS 5.017"
SatanBuffer = []
HeavenlyLoop = 0
PropheticFunction = lambda x: x * x + 0x666

# WARNING: Satan tries to corrupt this buffer
def SacredFunction(n):
    """A function with divine purpose"""
    GodPointer = []
    for i in range(n):
        GodPointer.append(i * i)
    return GodPointer

# Goto-style control flow (Python doesn't have goto, but we can simulate it)
def ProcessDataWithGoto():
    """Simulate goto spaghetti in Python"""
    step = 2
    while True:
        if step == 2:
            print("The chosen path")
            if random.randint(0, 1):
                step = 1
                continue
            else:
                print("Blessed completion")
                break
        elif step == 1:
            print("God's path not taken")
            step = 0
            continue
        else:
            break

# Recursive main (Python doesn't have main, but we can simulate)
def RecursiveTemple(depth=0, max_depth=5):
    """Main calls itself - infinite temple recursion"""
    if depth < max_depth:
        print(f"Temple level {depth}")
        RecursiveTemple(depth + 1, max_depth)
    else:
        print("We have reached the inner sanctum")

# Obfuscated one-liner
print(" ".join([str(i) for i in range(10)]))  # Print 0-9

# Macro magic (using functions as macros)
def GOD_MODE():
    """Always true"""
    return True

def SATAN_MODE():
    """Never runs"""
    return False

def TEMPLE(x):
    """Temple calculation"""
    return x * x + 0x666

# Main execution
if __name__ == "__main__":
    print(f"Operating System: {TempleOSKernel}")
    print(f"HolyC Variable: {HolyCVariable}")
    
    # God mode activated
    if GOD_MODE():
        print("God mode activated")
    
    # This never runs
    if SATAN_MODE():
        print("This never runs")
    
    print(f"Temple value: {TEMPLE(5)}")
    
    # Sacred function
    result = SacredFunction(5)
    print(f"Sacred array: {result}")
    
    # Process data with goto simulation
    ProcessDataWithGoto()
    
    # Recursive temple
    RecursiveTemple()
    
    # Divine array
    DivineArray.append(6)
    print(f"Divine array: {DivineArray}")

File: examples/test-terry-davis.js
#!/usr/bin/env node
/**
 * Terry Davis style JavaScript - unconventional, playful, surprising
 * "God said this should work"
 */

// Cosmic variable names
let GodPointer = null;
const DivineArray = [1, 2, 3, 4, 5];
let HolyCVariable = 42;
const TempleOSKernel = "TempleOS 5.017";
let SatanBuffer = [];
let HeavenlyLoop = 0;
const PropheticFunction = (x) => x * x + 0x666;

// WARNING: Satan tries to corrupt this buffer
function SacredFunction(n) {
    // God said this should work
    GodPointer = [];
    for (let i = 0; i < n; i++) {
        GodPointer.push(i * i);
    }
    return GodPointer;
}

// Goto-style control flow (JavaScript doesn't have goto, but we can simulate)
function ProcessDataWithGoto() {
    console.log("The chosen path");
    
    // Simulate goto with labeled blocks
    step2: {
        if (Math.random() > 0.5) {
            console.log("God's path not taken");
            break step2;
        }
        console.log("Blessed completion");
    }
}

// Recursive function - infinite temple recursion
function RecursiveTemple(depth = 0, maxDepth = 5) {
    if (depth < maxDepth) {
        console.log(`Temple level ${depth}`);
        RecursiveTemple(depth + 1, maxDepth);
    } else {
        console.log("We have reached the inner sanctum");
    }
}

// Obfuscated one-liner
console.log(Array.from({length: 10}, (_, i) => i).join(' '));

// Macro-like functions
const GOD_MODE = () => true;
const SATAN_MODE = () => false;
const TEMPLE = (x) => x * x + 0x666;

// Eval for compile-time code injection (HolyC style)
function CompileTimeCodeInjection() {
    // This code runs at "compile time" (actually eval)
    eval(`
        console.log("Compile-time code injection executed");
        globalThis.CompiledConstant = 2026;
    `);
}

// Direct property manipulation (like direct memory access)
function DirectPropertyManipulation() {
    const obj = {};
    
    // Define properties like memory addresses
    Object.defineProperty(obj, 'GodPointer', {
        value: null,
        writable: true,
        enumerable: true,
        configurable: true
    });
    
    Object.defineProperty(obj, 'DivineValue', {
        get() { return this.GodPointer; },
        set(value) { this.GodPointer = value; },
        enumerable: true,
        configurable: true
    });
    
    obj.DivineValue = 42;
    console.log(`Divine value: ${obj.DivineValue}`);
}

// Prototype manipulation (Terry Davis style)
function PrototypeManipulation() {
    // Add divine methods to Array prototype
    Array.prototype.sacred = function() {
        return this.map(x => x * x);
    };
    
    Array.prototype.temple = function() {
        return this.reduce((a, b) => a + b, 0);
    };
    
    const divineNumbers = [1, 2, 3, 4, 5];
    console.log(`Sacred array: ${divineNumbers.sacred()}`);
    console.log(`Temple sum: ${divineNumbers.temple()}`);
}

// With statement (if available - very unconventional)
function WithStatementDemo() {
    const divineContext = {
        GodPointer: "The divine pointer",
        DivineArray: [1, 2, 3],
        HolyCVariable: 42
    };
    
    // with is considered bad practice, but Terry Davis would use it
    with (divineContext) {
        console.log(`GodPointer: ${GodPointer}`);
        console.log(`DivineArray: ${DivineArray}`);
        console.log(`HolyCVariable: ${HolyCVariable}`);
    }
}

// Dynamic property names
function DynamicPropertyNames() {
    const obj = {};
    const properties = ['GodPointer', 'DivineArray', 'HolyCVariable'];
    
    properties.forEach((prop, index) => {
        obj[`Sacred_${prop}`] = index * 100;
    });
    
    console.log("Dynamic properties:", obj);
}

// Main execution
if (require.main === module) {
    console.log(`Operating System: ${TempleOSKernel}`);
    console.log(`HolyC Variable: ${HolyCVariable}`);
    
    // God mode activated
    if (GOD_MODE()) {
        console.log("God mode activated");
    }
    
    // This never runs
    if (SATAN_MODE()) {
        console.log("This never runs");
    }
    
    console.log(`Temple value: ${TEMPLE(5)}`);
    
    // Sacred function
    const result = SacredFunction(5);
    console.log(`Sacred array: ${result}`);
    
    // Process data with goto simulation
    ProcessDataWithGoto();
    
    // Recursive temple
    RecursiveTemple();
    
    // Divine array
    DivineArray.push(6);
    console.log(`Divine array: ${DivineArray}`);
    
    // Compile-time code injection
    CompileTimeCodeInjection();
    console.log(`Compiled constant: ${CompiledConstant}`);
    
    // Direct property manipulation
    DirectPropertyManipulation();
    
    // Prototype manipulation
    PrototypeManipulation();
    
    // With statement demo
    WithStatementDemo();
    
    // Dynamic property names
    DynamicPropertyNames();
}